### DISCUSSION TITLE
Possible to change the cidata.iso?

### DISCUSSION THREAD

**Discussion by oilcan-productions** [lima-vm/lima#794] [Q&A] [answered]
I would like to launch a custom Linux image and have already a cloudint based ISO for that. Can I merge that content into the cidata.iso or vice-versa without breaking something? Is there more docs on what is in the cidata.iso?
Looking at the lifecycle diagram it looks like the ISO is created on the fly.
Is there documentation on how to add custom code to that process so that my own cloud init parts are added as well rather than replacing the ISO on the fly once it is copied over?

**Comment by maxammann** [comment]
I suspect this is not in the goals of lima. But multipass supports this. Though it only officially support ubuntu.

**Comment by AkihiroSuda** [comment, marked as answer]
No plan to support replacing `cidata.iso`, but you can set a custom provisioning script in the yaml
https://github.com/lima-vm/lima/blob/5d9221a3912d9026a682df37cc1b29d92ff9c0be/examples/apptainer.yaml#L21-L30
